From 3c056b56dac7609e596a5b73f2acb738c45372c2 Mon Sep 17 00:00:00 2001
From: Liu Ying <b17645@freescale.com>
Date: Wed, 1 Dec 2010 16:06:09 +0800
Subject: [PATCH 196/449] ENGR00134203 TVin:Support more test options

1) Support to choose bottom/top field first.
2) Support to choose VDI motion.

Signed-off-by: Liu Ying <b17645@freescale.com>
---
 test/mxc_v4l2_test/mxc_v4l2_tvin.c | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/test/mxc_v4l2_test/mxc_v4l2_tvin.c b/test/mxc_v4l2_test/mxc_v4l2_tvin.c
index ca8abc9..14599a6 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_tvin.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_tvin.c
@@ -60,6 +60,8 @@ int g_cap_mode = 0;
 int g_input = 1;
 int g_fmt = V4L2_PIX_FMT_UYVY;
 int g_rotate = 0;
+int g_motion = 0;
+int g_tb = 0;
 int g_output = 3;
 int g_output_num_buffers = 4;
 int g_capture_num_buffers = 3;
@@ -278,7 +280,7 @@ int v4l_output_setup(void)
         }
 
 	ctrl.id = V4L2_CID_PRIVATE_BASE + 3;
-        ctrl.value = 0;
+        ctrl.value = g_motion;
         if (ioctl(fd_output_v4l, VIDIOC_S_CTRL, &ctrl) < 0)
         {
                 printf("set ctrl failed\n");
@@ -296,7 +298,10 @@ int v4l_output_setup(void)
 	fmt.fmt.pix.bytesperline = g_in_width;
 	fmt.fmt.pix.priv = 0;
 	fmt.fmt.pix.sizeimage = 0;
-	fmt.fmt.pix.field = V4L2_FIELD_INTERLACED_BT;
+	if (g_tb)
+		fmt.fmt.pix.field = V4L2_FIELD_INTERLACED_TB;
+	else
+		fmt.fmt.pix.field = V4L2_FIELD_INTERLACED_BT;
         if (ioctl(fd_output_v4l, VIDIOC_S_FMT, &fmt) < 0)
         {
                 printf("set format failed\n");
@@ -503,6 +508,12 @@ int process_cmdline(int argc, char **argv)
 				printf("Default format is used: UYVY\n");
                         }
                 }
+                else if (strcmp(argv[i], "-m") == 0) {
+                        g_motion = atoi(argv[++i]);
+                }
+                else if (strcmp(argv[i], "-tb") == 0) {
+                        g_tb = 1;
+                }
                 else if (strcmp(argv[i], "-help") == 0) {
                         printf("MXC Video4Linux TVin Test\n\n" \
 			       "Syntax: mxc_v4l2_tvin.out\n" \
@@ -511,6 +522,8 @@ int process_cmdline(int argc, char **argv)
 			       " -ot <display top>\n" \
 			       " -ol <display left>\n" \
                                " -r <rotation> -c <capture counter> \n"
+			       " -m <motion> 0:medium 1:low 2:high, 0-default\n"
+			       " -tb top field first, bottom field first-default\n"
 			       " -f <format, only YU12, YUYV, UYVY and NV12 are supported> \n");
                         return TFAIL;
                }
-- 
1.8.0

